build/ci.go

```diff
@@ -465,6 +496,63 @@ func stageDebianSource(tmpdir string, meta debMetadata) (pkgdir string) {
 	return pkgdir
 }
 
+// Azure binary hosting
+
+type azureBlobstore struct {
+	account   string // Account name to authorize API requests with
+	token     string // Access token for the above account
+	container string // Blob container to upload files into
+}
+
+// azureUpload uploads a local file to the Azure Blob Storage. Note, this method
+// assumes a max file size of 64MB (Azure limitation). Larger files will need a
+// multi API call approach implemented.
+//
+// See: https://msdn.microsoft.com/en-us/library/azure/dd179451.aspx#Anchor_3
+func azureUpload(path string, name string, store azureBlobstore) error {
+	// Create an authenticated client against the Azure cloud
+	rawClient, err := storage.NewBasicClient(store.account, store.token)
+	if err != nil {
+		return err
+	}
+	client := rawClient.GetBlobService()
+
+	// Stream the file to upload into the designated blobstore container
+	in, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+	defer in.Close()
+
+	info, err := in.Stat()
+	if err != nil {
+		return err
+	}
+	return client.CreateBlockBlobFromReader(store.container, name, uint64(info.Size()), in, nil)
+}
+
+// signFile reads the contents of an input file and signs it (in armored format)
+// with the key provided, placing the signature into the output file.
+func signFile(input string, output string, pgpkey string) error {
+	keys, err := openpgp.ReadArmoredKeyRing(bytes.NewBufferString(pgpkey))
+	if err != nil {
+		return err
+	}
+	in, err := os.Open(input)
+	if err != nil {
+		return err
+	}
+	defer in.Close()
+
+	out, err := os.Create(output)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	return openpgp.ArmoredDetachSignText(out, keys[0], in, nil)
+}
```

Comment:
Please put these two functions into the internal/build package. They are kind of standalone.